token/client/src/token.rs

```diff
@@ -377,7 +432,7 @@ where
     /// Create and initialize a new token account.
     pub async fn create_auxiliary_token_account(
         &self,
-        account: &S,
+        account: &dyn Signer,
```

Comment:
id made this `Arc<dyn Signer>` but it broke every test, but this function is never used outside of tests so we can beautify this later. i feel like theres some cute shit that could be done with `AsRef`